python/lsst/pipe/base/graphBuilder.py

```diff
@@ -817,7 +819,79 @@ def makeGraph(self, pipeline, collections, run, userQuery):
             classes.
         """
         scaffolding = _PipelineScaffolding(pipeline, registry=self.registry)
+
+        userQuery = self._addInstrumentToQuery(pipeline, userQuery)
+
         with scaffolding.connectDataIds(self.registry, collections, userQuery) as commonDataIds:
             scaffolding.resolveDatasetRefs(self.registry, collections, run, commonDataIds,
                                            skipExisting=self.skipExisting)
         return scaffolding.makeQuantumGraph()
+
+    def _addInstrumentToQuery(self, pipeline, query):
+        """Add an instrument restriction to the query if the pipeline has an
+        instrument and the query does not already have an instrument. If the
+        query does have an instrument verify that it matches the instrument in
+        the pipeline.
+
+        Parameters
+        ----------
+        pipeline : `Pipeline`
+            Same as the pipeline in `makeGraph`
+        query : `str`
+            The initial query string.
+
+        Returns
+        -------
+        query : `str`
+            The query string with an instrument restriction checked or added
+            if needed.
+
+        Raises
+        ------
+        RuntimeError
+            If the query contains an instrument restriction and the pipeline
+            has an instrument and the two do not match.
+        """
+        instrument = pipeline.getInstrument()
+        if instrument is not None:
+            if isinstance(instrument, str):
+                instrument = getInstrument(pipeline._pipelineIR.instrument, self.registry)
+            queryInstrument = self._getInstrumentFromQuery(query)
+            if queryInstrument is None:
+                # There is not an instrument in the query, add it:
+                restriction = f"instrument = '{instrument.getName()}'"
+                _LOG.info(f"Adding restriction \"{restriction}\" to query.")
```

Comment:
Maybe `debug`, that is not super-useful as INFO, unless you like to annoy people 🙂